scaleway/data_source_vpc_private_network.go

```diff
@@ -50,9 +57,11 @@ func dataSourceScalewayVPCPrivateNetworkRead(ctx context.Context, d *schema.Reso
 			return diag.FromErr(err)
 		}
 
```

Comment:
Please add the vpc_id directly in the list request instead

```suggestion
res, err := vpcAPI.ListPrivateNetworks(
			&vpc.ListPrivateNetworksRequest{
				Name:      expandStringPtr(pnName),
				Region:    region,
				ProjectID: expandStringPtr(d.Get("project_id")),
				VpcID: expandStringPtr(d.Get("vpc_id")),
			}, scw.WithContext(ctx))
```